wagtail/wagtailcore/models.py

```diff
@@ -1649,6 +1703,32 @@ def publishable_pages(self):
 
         return publishable_pages
 
+    def permitted_paths(self, choosable=False):
+        """
+        Return a list of paths to Pages that this User is directly permitted to explore/choose.
+        This is not the same as the *_pages functions in this class, as it only returns paths to
+        the pages which are directly permissioned in the GroupPagePermissions model. It does not return
+        decendant paths of said pages.
+        Use choosable=True to return paths for Pages that are choosable, rather than Pages that are explorable.
+        """
+        if not self.user.is_active:
+            return []
+        if self.user.is_superuser:
+            # Superusers are permitted to explore the tree root, which lets them explore every Page on the server.
+            return [Page.get_first_root_node().path]
+
+        # Retrieve the path(s) for each Page for which this user has either at least one GroupPagePermission
+        # that provides explorability, or for which the user has the 'choose' GroupPagePermission.
+        permitted_paths = Page.objects.distinct().values_list('path', flat=True).filter(
+            group_permissions__group__in=self.user.groups.all()
+        )
+        if choosable:
+            permitted_paths = permitted_paths.filter(group_permissions__permission_type='choose')
```

Comment:
This permission type doesn't exist (yet...), so as it stands this has the effect of removing the breadcrumb entirely on the page chooser for non-superusers.